tuf/client_rework/updater_rework.py

```diff
@@ -35,27 +36,26 @@ class Updater:
     """
     Provides a class that can download target files securely.
 
-    Attributes:
-        metadata:
-
-        repository_name:
-
-        mirrors:
-
-        fetcher:
-
-        consistent_snapshot:
+    TODO
     """
 
     def __init__(
         self,
         repository_name: str,
-        repository_mirrors: Dict,
+        metadata_url: str,
+        default_target_url: Optional[str] = None,
         fetcher: Optional[FetcherInterface] = None,
     ):
 
         self._repository_name = repository_name
-        self._mirrors = repository_mirrors
+        self._metadata_url = metadata_url
+        # Should we accept metadata url as a default for targets or
+        # targets_url should be provided either in this constructor
+        # or as a download_target parameter?
```

Comment:
I think it makes sense to default to using the same host/url combination for all downloads, unless told otherwise via the `targets_base_url` or the optional function parameters.

Reply:
I don't think it works though: metadata base url directory should contain metadata (it's like current `url_prefix+metadata_path`). I would be surprised if someone wants to store targets in the same directory. I think it's reasonable to expect either a default target url prefix or the individual target download url prefixes

Reply:
Jussi, I agree with you, but I think Joshua simply means the following: there is a `metadata_base_url` parameter that must always be explicitly set. However, there is also a `targets_base_url` that defaults to `metadata_base_url` unless also explicitly set to something else.

Reply:
I think I understood: I'm just wondering if anyone would want it to default to metadata_base_url?